blog/assembly/contracts/main.ts

```diff
@@ -0,0 +1,53 @@
+import { Storage, callerHasWriteAccess, generateEvent } from '@massalabs/massa-as-sdk';
+import { Args } from '@massalabs/as-types';
+
+/**
+ * This function is meant to be called only one time: when the contract is deployed.
+ *
+ * @param _ - not used
+ */
+export function constructor(_: StaticArray<u8>): StaticArray<u8> {
+  // This line is important. It ensures that this function can't be called in the future.
+  // If you remove this check, someone could call your constructor function and reset your smart contract.
+  if (!callerHasWriteAccess()) {
```

Comment:
You should be able to use [isDeployingContract](https://github.com/massalabs/massa-as-sdk/blob/main/assembly/std/context.ts#L46) which is cleaner

Reply:
Why was this discussioin resolved ? isDeployingContract is not working ?